### PR TITLE
Limit printing too many blocks for MasterWorkerInfo toString()

### PR DESCRIPTION
**Pull request by jiacheliu3**
This makes `MasterWorkerInfo` only prints no more than 99 blocks instead of the whole list. The whole list can be incredibly long like thousands of blocks.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java [hunk 1]
```diff
@@ -375,9 +376,13 @@ public boolean hasLostStorage() {
 
   @Override
   public String toString() {
+    int blockSizeLimit = 100;
     return MoreObjects.toStringHelper(this).add("id", mId).add("workerAddress", mWorkerAddress)
         .add("capacityBytes", mCapacityBytes).add("usedBytes", mUsedBytes)
-        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).add("blocks", mBlocks)
+        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs)
+        .add("blocks",
+              (mBlocks.size() < blockSizeLimit) ? mBlocks :
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:384]: you shouldn't need this condition with `limit`.

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java [hunk 2]
```diff
@@ -375,9 +376,13 @@ public boolean hasLostStorage() {
 
   @Override
   public String toString() {
+    int blockSizeLimit = 100;
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:379]: I feel like 100 is pretty low for this?

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java [hunk 3]
```diff
@@ -375,9 +376,13 @@ public boolean hasLostStorage() {
...
     return MoreObjects.toStringHelper(this).add("id", mId).add("workerAddress", mWorkerAddress)
         .add("capacityBytes", mCapacityBytes).add("usedBytes", mUsedBytes)
-        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).add("blocks", mBlocks)
+        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs)
+        .add("blocks",
+              (mBlocks.size() < blockSizeLimit) ? mBlocks :
+                mBlocks.stream().limit(blockSizeLimit).collect(Collectors.toList()))
         .add("lostStorage", mLostStorage).toString();
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:386]: now that we're truncating the list, can we add block-count as a separate field?

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java (hunk 1 above)
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:384]: I think adding this condition will reduce one copy if we don't need to truncate?

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java (hunk 2 above)
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:379]: This is a good question. I've no idea what is a good split and don't have statistics for the decision. I can only say a MasterWorkerInfo of a good 400 blocks can fill my 15-inch screen. 

The benefit of not truncating this list is we can search for a specific block id and know for sure if it exists on one worker. Once we start truncating there's no such nice certainty. So if we do truncate, intuitively I lean towards using a smaller limit, as we lose the certainty whatsoever.

Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:379]: Can we do it like this, truncate in normal mode, and leave it as is if debug enabled.
@gpang WDYT about this in general?

Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:380]: Good idea!

**Comment by ZacBlanco** [comment]
@ggezer @jiacheliu3 I would like to merge this. Can we finish the review?

**Comment by ggezer** [comment]
@jiacheliu3 Can you open an issue linked to this PR. I don't think we should be cutting info but rather think this through later.

**Comment by jiacheliu3** [comment]
@ggezer https://github.com/Alluxio/alluxio/issues/11729 logged for this topic

**Comment by ZacBlanco** [comment]
ping!

**Comment by jiacheliu3** [comment]
I'll close this PR as we need to further think it through (on the ticket).
(pong)

**Pull request closed without merging**